primer/test/Tests/EvalFull.hs

```diff
@@ -1692,6 +1693,20 @@ unit_prim_ann =
         s <- evalFullTest maxID builtinTypes prims 2 Syn e
         s <~==> Right r
 
+unit_prim_lazy :: Assertion
+unit_prim_lazy =
+  let ((e, r, prims), maxID) =
+        create $
+          (,,)
+            <$> pfun PrimConst
+            `app` bool_ True
+            `app` emptyHole
```

Comment:
Could we also have a test that `const` does not force its second argument? i.e. with something like `letrec x = x : Nat in x` in place of this `emptyHole`?

Reply:
Sure, now added (I was thinking it didn't really add much over the first test, but I guess it does make sure we don't eagerly go off and prioritise expanding `let`s).